contracts/Comptroller/Diamond/facets/PolicyFacet.sol

```diff
@@ -28,6 +47,8 @@ contract PolicyFacet is IPolicyFacet, XVSRewardsHelper {
      * @return 0 if the mint is allowed, otherwise a semi-opaque error code (See ErrorReporter.sol)
      */
     function mintAllowed(address vToken, address minter, uint256 mintAmount) external returns (uint256) {
+        ensureUserAllowed(minter);
```

Comment:
I think we should also check if the caller is allowed to mint tokens on behalf of the minter
```suggestion
        ensureUserAllowed(minter);
        ensureUserAllowed(msg.sender);
```